### ISSUE TITLE
_pad_circular incorrect padding when pads[-1], pads[-3], or pads[-5] = 0

### ISSUE THREAD

**Issue by millskyle** [pytorch/pytorch#37256]
## To Reproduce

1. Call _pad_circular with pads=[1,0,2,2]
2. Output shape will be incorrect.

This is because the indexing of the line like

`input = torch.cat([input[:, :, -(padding[-1] + padding[-2]):-padding[-1]], input], dim=2`

 doesn't work if `padding[-1] = 0` as then 

`-(padding[-1] + padding[-2]):-padding[-1]` = `-padding[-2]:-0`

when it should actually be `-padding[-2]:`

**Comment by mruberry** [comment]
I believe this was addressed by https://github.com/pytorch/pytorch/pull/39273.

@millskyle, would you verify?

**Comment by mruberry** [comment]
Closing this due to age and lack of an update. If this issue is still occurring please reopen it, @millskyle.
